api/api/serializers/media_serializers.py

```diff
@@ -628,37 +684,56 @@ class MediaSearchRequestSourceSerializer(serializers.Serializer):
         Keep the fields names in sync with the actual fields below as this list is
         used to generate Swagger documentation.
         """
-
-        _field_attrs = {
-            "help_text": (
-                "A comma separated list of data sources; valid values are "
-                "``source_name``s from the stats endpoint: "
-                f"{settings.CANONICAL_ORIGIN}/v1/{media_path}/stats/."
-            ),
-            "required": False,
+        source_help_text = SOURCE_HELP_TEXT % {
+            "origin": settings.CANONICAL_ORIGIN,
+            "media_path": media_path,
+        }
+        excluded_help_text = EXCLUDED_SOURCE_HELP_TEXT % {
+            "origin": settings.CANONICAL_ORIGIN,
+            "media_path": media_path,
         }
 
         source = serializers.CharField(
             label="provider",
-            **_field_attrs,
+            help_text=source_help_text,
+            required=False,
         )
         excluded_source = serializers.CharField(
             label="excluded_provider",
-            **_field_attrs,
+            help_text=excluded_help_text,
+            required=False,
         )
 
-        @staticmethod
-        def validate_source_field(value):
-            """Check whether source is a valid source."""
-
+        def validate_source(self, value):
+            """
+            For the regular searches, split the value.lower() by comma and only return the
+            source names that are in the search controller's sources list for the media type.
+            For the collection=tag, return the value as is. It is ignored in the query builder.
+            For source and creator collections, accept the value as is, without lower-casing
+            or splitting, and check if it's in the valid source name list.
+            This function validates the source and excluded_source fields, but `excluded_source`
+            is ignored for collection requests.
+            """
             allowed_sources = list(search_controller.get_sources(media_type).keys())
-            sources = value.lower().split(",")
-            sources = [source for source in sources if source in allowed_sources]
-            value = ",".join(sources)
-            return value
-
-        def validate_source(self, input_sources):
-            return self.validate_source_field(input_sources)
+            sources_list = ", ".join([f"'{s}'" for s in allowed_sources])
+            collection = self.initial_data.get(COLLECTION)
+
+            # For collection=tag, return the value as is. It is ignored in the query builder.
+            if collection == "tag":
+                return value
```

Comment:
Shouldn't this raise a validation error, as they are not compatible parameters, rather than just ignored? I'm having a hard time keeping straight what we're ignoring vs validating... spreading these out between separate serializers (MediaSearchRequestSourceSerializer and the regular search request serializer) is making it hard to keep track of what is validating what and where!

Are we able to reduce down to a single serializer, by chance? Or could the validation be built into a new `SourceField` that validates differently based on the strategy?

Reply:
> MediaSearchRequestSourceSerializer and the regular search request serializer

These two serializers were separate before any additional search views project work. The reason for that, as I understand, is that `media_type` is not available in the regular media serializer, and it's necessary for the source to check the relevant source list (audio sources for audio and image sources for image). The source serializer is dynamic.